src/sammich/plugins/project.py

```diff
@@ -1,32 +1,91 @@
 import json
+import re
 
 from machine.clients.slack import SlackClient
 from machine.plugins.base import MachineBasePlugin
-from machine.plugins.decorators import command
+from machine.plugins.decorators import action, command
 from machine.storage import PluginStorage
 from machine.utils.collections import CaseInsensitiveDict
 
+PROJECTS_PER_PAGE = 100
+
 
 class ProjectPlugin(MachineBasePlugin):
     def __init__(self, client: SlackClient, settings: CaseInsensitiveDict, storage: PluginStorage):
         super().__init__(client, settings, storage)
-        with open("data/projects.json") as f:
-            self.project_data = json.load(f)
+
+        try:
+            with open("data/projects.json") as f:
+                self.project_data = json.load(f)
+        except json.JSONDecodeError:
+            self.project_data = {}
 
     @command("/project")
     async def project(self, command):
-        text = command.text.strip()
-        project_name = text.strip().lower()
+        project_name = command.text.strip().lower()
+        channel_id = command._cmd_payload["channel_id"]
 
         project = self.project_data.get(project_name)
 
         if project:
             project_list = "\n".join(project)
             message = f"Hello, here the information about '{project_name}':\n{project_list}"
+            await command.say(message)
         else:
-            message = (
-                f"Hello, the project '{project_name}' is not recognized. "
-                "Please try different query."
+            # if the self.project_data is empty, return a message
+            if not self.project_data:
+                await command.say("No projects available")
+                return None
+            await self.show_project_page(channel_id)
+
+    async def show_project_page(self, channel_id):
+        projects = list(self.project_data.keys())
+
+        # Calculate the number of dropdowns needed
+        num_dropdowns = (len(projects) + PROJECTS_PER_PAGE - 1) // PROJECTS_PER_PAGE
+
+        blocks = []
+        for i in range(num_dropdowns):
+            project_for_each_page = 75
```

Comment:
I suggest moving this out of the loop.

Reply:
ok sure , I didn't notice that , very sorry